superset/connectors/druid/models.py

```diff
@@ -1148,6 +1148,11 @@ def run_query(  # noqa / druid
                     metric['column']['type'].upper() == 'FLOAT'
                 ):
                     metric['column']['type'] = 'DOUBLE'
+            if (
+                utils.is_adhoc_metric(timeseries_limit_metric) and
+                timeseries_limit_metric['column']['type'].upper() == 'FLOAT'
```

Comment:
I think you also need to add a version check here, see:

https://github.com/apache/incubator-superset/pull/5030/files

Reply:
unless this is a different thing?

Reply:
The version check is happening a few lines above. https://github.com/apache/incubator-superset/pull/5118/files#diff-a8dd5ec8d8decda2e3c5571d1ec0cdb6R1141

Reply:
O i see

Reply:
It's almost the exact same code as a few lines above, maybe refactor a `sanitize_metric_object` and put as much as possible in there.

[probably out of scope for this PR] I'm starting to think we should make all metrics fit the same model (make predefined metrics look like custom metric objects) as early as possible in the stack and then treat them all the same way.